openpype/hosts/blender/plugins/publish/validate_no_colons_in_name.py

```diff
@@ -12,20 +15,21 @@ class ValidateNoColonsInName(pyblish.api.InstancePlugin):
 
     """
 
-    order = openpype.api.ValidateContentsOrder
+    order = ValidateContentsOrder
     hosts = ["blender"]
     families = ["model", "rig"]
+    category = "cleanup"
     version = (0, 1, 0)
     label = "No Colons in names"
-    actions = [openpype.hosts.blender.api.action.SelectInvalidAction]
+    actions = [SelectInvalidAction]
 
-    @classmethod
+    @staticmethod
     def get_invalid(cls, instance) -> List:
         invalid = []
-        for obj in [obj for obj in instance]:
+        for obj in set(instance):
             if ':' in obj.name:
                 invalid.append(obj)
-            if obj.type == 'ARMATURE':
+            if isinstance(obj, bpy.types.Object) and obj.type == 'ARMATURE':
                 for bone in obj.data.bones:
                     if ':' in bone.name:
                         invalid.append(obj)
```

Comment:
I have no knowledge about how the blender logic works for OpenPype. But in Maya terminology the "history" of the instance (like its bones for a rig) are included into the `instance` as nodes/objects. Wouldn't that make more sense here too? So that we wouldn't have to traverse into the data armatures to begin with and collect its history / armature bones in a collector? (Again, not a comment about the code change - sorry!)

Reply:
Hi ! Blender has very atypical way to construct rigs. Bones are not objects but rather sub-elements of an armature type object. But this could be possible to append `bpy.type.Bone` in `collect_instances.py` as members now that we check members classes as `bpy.types.Object` in all Validators how needed it.

Reply:
I see - if they aren't objects like that I can see how this might not be the most sensible way to embed it as history. Thanks.

Reply:
Yes actually Extractors could raise an `AttributeError` if we add `bpy.types.Bone`: [extract_blend.py#L31](https://github.com/pypeclub/OpenPype/blob/develop/openpype/hosts/blender/plugins/publish/extract_blend.py#L31)